src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -319,6 +355,41 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
             }
         }
 
+        private static void PopulateReferenceIdDictionary(IEnumerable<EntryComponent> bundleEntries, IDictionary<string, (string resourceId, string resourceType)> idDictionary)
+        {
+            foreach (EntryComponent entry in bundleEntries)
+            {
+                if (entry.Request.Method != HTTPVerb.POST)
+                {
+                    continue;
+                }
+
+                // We've already come across this ID
+                if (!idDictionary.TryGetValue(entry.FullUrl, out _))
```

Comment:
Is this just ContainsKey?:
```suggestion
                if (!idDictionary.ContainsKey(entry.FullUrl))
```